src/App.jsx

```diff
@@ -14,33 +16,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortOrder, setSortOrder] = useState(null);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortGoods = order => {
+    let sortedGoods;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    switch (order) {
+      case 'alphabet':
+        sortedGoods = [...goodsFromServer].sort();
+        break;
+      case 'length':
+        sortedGoods = [...goodsFromServer].sort((a, b) => a.length - b.length);
+        break;
+      default:
+        sortedGoods = [...goodsFromServer];
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (isReversed) sortedGoods.reverse();
+    setGoods(sortedGoods);
+    setSortOrder(order);
+  };
+
+  const reverseGoods = () => {
+    setGoods([...goods].reverse());
+    setIsReversed(!isReversed);
```

Comment:
[functional setState](https://react.dev/reference/react/useState)?

```suggestion
    setGoods((prevGoods) => [...prevGoods].reverse());
    ....
```
but after u moved the function outside - u will have to remove this functionality anyway)

Reply:
not fixed!
you can update state based on the previous state

https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state